src/ast.c

```diff
@@ -1046,6 +1046,125 @@ int has_meta(jl_array_t *body, jl_sym_t *sym)
     return 0;
 }
 
+void renumber_slots(jl_array_t *code, size_t *newslot);
+
+jl_lambda_info_t *desplat_lambda(jl_lambda_info_t *li, jl_tupletype_t *types)
+{
+    assert(li->isva);
+    size_t *newslot = (size_t*) malloc(jl_array_len(li->slotnames)*sizeof(size_t)+1);
+    jl_array_t *code = li->code;
+    int32_t nargs = li->nargs;
+    size_t nslurp = jl_nparams(types)-nargs+1;
+    jl_array_t *newa = NULL;
+    jl_expr_t *slurpexpr = NULL;
+    jl_expr_t *subexpr = NULL;
+    jl_lambda_info_t *newmeth = jl_copy_lambda_info(li);
+    size_t i;
+    JL_GC_PUSH5(&newa, &newmeth, &slurpexpr, &subexpr, &code);
+    if (!jl_typeis(code,jl_array_any_type)) {
+        code = jl_uncompress_ast(li, code);
+    }
+    // Signature: replace slurping argument with explicit arguments
+    int vapos = nargs-1;
+    // slotnames
+    jl_sym_t *vaname = (jl_sym_t*)jl_cellref(newmeth->slotnames,vapos);
+    newa = jl_alloc_array_1d(jl_array_any_type, jl_array_len(newmeth->slotnames)+nslurp);
+    jl_gc_wb(newmeth, newa);
```

Comment:
You want `jl_gc_wb` when you assign to a field or array. It is not necessary here.